### ISSUE TITLE
PF_RING error when compiling

### ISSUE THREAD

**Issue by hamidrezabstn** [google/gopacket#419]
when compile my project this error shows

`github.com/google/gopacket/pfring
../../pfring/pfring.go:155: r.pkthdr.extended_hdr.if_index undefined (type C.struct_pfring_extended_pkthdr has no field or method if_index)`

**Comment by florianl** [comment]
Can you provide a minimal working code example,that produces the error?

**Comment by florianl** [comment]
In the used library I still can find the field id_index at 
https://github.com/ntop/PF_RING/blob/5660e0c69648e41a0beb4c58969956372753d4f1/kernel/linux/pf_ring.h#L317

**Comment by hamidrezabstn** [comment]
hi
i've installed PF_RING using their official tutorial : [Tutorial](https://www.ntop.org/pf_ring/installation-guide-for-pf_ring/)
and inside httpassembly  example of gopacket i just include pfring and replace it with pcap! 
and when i try to compile it i see this error! 
i also check  my pf_ring.h file and it has that property just like you have said but still i see that error

**Comment by florianl** [comment]
hi @hamidrezabstn 
it would be great, if you could provide a minimal working code example, that causes this error. In addition, this error sounds for me like a dependency problem.

**Comment by hamidrezabstn** [comment]
hi @florianl 
as i've said  "inside httpassembly example of gopacket i just include pfring and replace it with pcap!"
[example file](https://github.com/google/gopacket/tree/master/examples/httpassembly)

**Comment by florianl** [comment]
hi @hamidrezabstn 
maybe I missunderstood you. If I would use pfring inside the httpassembly example, it would look like [1] (this is not tested and may contain errors, as I don't have PF_RING installed)

Can you run the examples from the PR_FING library [2]?

[1] https://gist.github.com/florianl/d291f08c3a76b0a393c870ff34de7f33
[2] https://github.com/ntop/PF_RING/tree/dev/userland/examples

**Comment by ggmartins** [comment]
+1
```
$go build gistfile1.go
# github.com/google/gopacket/pfring
src/github.com/google/gopacket/pfring/pfring.go:155: r.pkthdr.extended_hdr.if_index undefined (type _Ctype_struct_pfring_extended_pkthdr has no field or method if_index)
```

```
$go get github.com/google/gopacket/pfring
# github.com/google/gopacket/pfring
src/github.com/google/gopacket/pfring/pfring.go:155: r.pkthdr.extended_hdr.if_index undefined (type _Ctype_struct_pfring_extended_pkthdr has no field or method if_index)
```

```
$sudo ./pfcount 
Using PF_RING v.7.1.0
Capturing from eth0 [mac: B8:27:EB:4E:74:A1][if_index: 2][speed: 100Mb/s]
# Device RX channels: 1
# Polling threads:    1
Dumping statistics on /proc/net/pf_ring/stats/9466-eth0.2
=========================
Absolute Stats: [520 pkts total][0 pkts dropped][0.0% dropped]
[520 pkts rcvd][385'048 bytes rcvd]
...
```

**Comment by lightinbottle** [comment]
+1
i had the same problem.
somebody has a solution?

**Comment by solidgoldbomb** [comment]
If you are trying to build gopacket/pfring on a system that has PF_RING v7.2.0 installed then your compile issues may be explained by this: https://github.com/google/gopacket/issues/467#issuecomment-408901729.

Basically, v7.2.0 marked one of their structs as packed.  That struct contains unaligned fields.  cgo deliberately hides those unaligned fields from go code.  That leads to a failure to compile the go code even though the field is clearly still in the C header file.

@florianl this is going to hit more users as they upgrade to upstream PF_RING v7.2.0 and try to build gopacket/pfring.  The issue is readily reproducible with an empty (or bypassed) go cache when you build any golang program that imports gopacket/pfring.

**Comment by notti** [comment]
Fixed by #553
